lib/sensu/client/http_socket.rb

```diff
@@ -0,0 +1,208 @@
+    # the following endpoints:
+    #
+    # GET /info
+    #  This endpoint returns 200 OK with some basic sensu info
+    #
+    # POST /results
+    #  This endpoint expects application/json body with a check result
+    #
+    # GET /settings
+    #  This endpoint responds with 200 OK and the sensu configuration
+    #
+    # GET /brew
+    #  This endpoint gets you some fresh coffee
+    class HTTPSocket < EM::HttpServer::Server
+      include Sensu::API::Utilities::TransportInfo
+      include Utilities
+      include CheckUtils
+
+      attr_accessor :logger, :settings, :transport
+
+      def initialize
+        super
+        @endpoints = {
+          "/info" => {
+            "methods" => {
+              "GET" => method(:process_request_info)
+            },
+            "help" => "Sensu client information"
+          },
+          "/results" => {
+            "methods" => {
+              "POST" => method(:process_request_results)
+            },
+            "help" => "Send check json results here"
+          },
+          "/settings" => {
+            "methods" => {
+              "GET" => method(:process_request_settings)
+            },
+            "help" => "Get sensu settings (requires basic auth)"
+          },
+          "/brew" => {
+            "methods" => {
+              "GET" => Proc.new { |response|
+                response.status = 418
+                response.status_string = "I'm a teapot"
+                response.content = Sensu::JSON::dump({:response => "I'm a teapot!"})
+                response.send_response
+              }
+            },
+            "help" => "Ask Sensu to brew a cup of joe (try it!)"
+          }
+        }
+      end
+
+      def authorized?
+        http_options = @settings[:client][:http_socket] || Hash.new
+        if http_options[:user] and http_options[:password]
+          if @http[:authorization]
+            scheme, base64 = @http[:authorization].split("\s")
+            if scheme == "Basic"
+              user, password = Base64.decode64(base64).split(":")
+              return (user == http_options[:user] && password == http_options[:password])
+            end
+          end
+        end
+        false
+      end
+
+      def process_request_info(response)
+        @logger.debug("Processing info request")
+        transport_info do |info|
+          rdata = {
+            :sensu => {
+              :version => VERSION
+            },
+            :transport => info
+          }
+          response.content = Sensu::JSON::dump(rdata)
+          response.send_response
+        end
+      end
+
+      def process_request_results(response)
+        @logger.debug("Processing results request")
+        if @http[:content_type] and @http[:content_type] == 'application/json' and @http_content
+          begin
+            check = Sensu::JSON::load(@http_content)
+            process_check_result(check)
+            rdata = {
```

Comment:
I can do much better, wait for my update, it'll remove much duplication and looks cleaner IMO